pkg/engine/diff.go

```diff
@@ -508,8 +508,7 @@ func printPropertyValueDiff(
 
 		if shouldPrintOld && shouldPrintNew {
 			if diff.Old.IsArchive() &&
-				diff.New.IsArchive() &&
-				!causedReplace {
+				diff.New.IsArchive() {
```

Comment:
note: this !causedReplace line was something that was here since some of the earliest days of pulumi.  I never touched it because i wasn't certain what impact it might have.  Now that i've run into this issue first hand, i def think this is something we do *not* want, and we should essentially always print two archives as a diff, regardless of what the cause was.